File: onebot/src/main/java/cn/evole/onebot/sdk/response/misc/UserInfoResp.java
package cn.evole.onebot.sdk.response.misc;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Created on 2022/12/6.
 *
 * @author MrXiaoM
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserInfoResp {

    @SerializedName("user_id")
    long uin;
    @SerializedName("user_name")
    String name;
    @SerializedName("user_displayname")
    String displayName;
    @SerializedName("user_remark")
    String remark;
    @SerializedName("mail")
    String mail;
    @SerializedName("find_method")
    String findMethod;
    @SerializedName("max_vote_cnt")
    short maxVoteCnt;
    @SerializedName("have_vote_cnt")
    short haveVoteCnt;
    @SerializedName("vip_list")
    List<VipInfo> vipList;
    @SerializedName("hobby_entry")
    String hobbyEntry;
    @SerializedName("level")
    int level;
    @SerializedName("birthday")
    long birthday;
    @SerializedName("login_day")
    long loginDay;
    @SerializedName("vote_cnt")
    long voteCnt;
    @SerializedName("qid")
    String qid;
    @SerializedName("is_school_verified")
    Boolean schoolVerified;
    @SerializedName("location")
    Location location;
    @SerializedName("cookie")
    byte[] cookie;
    @Data
    public static class VipInfo {
        /**
         * QQ_VIP,
         * SUPER_QQ,
         * SUPER_VIP,
         * QQ_VIDEO,
         * QQ_READING,
         * BIG_VIP,
         * YELLOW_VIP
         */
        String type;
        int level;
        @SerializedName("vip_type")
        int vipType;
        @SerializedName("template_id")
        long templateId;
    }
    @Data
    public static class Location {
        String city;
        String company;
        String country;
        String province;
        String hometown;
        String school;
    }
}

File: onebot/src/main/kotlin/sdk/response/misc/UserInfoResp.kt
package cn.evolvefield.onebot.sdk.response.misc

import com.google.gson.annotations.SerializedName
import lombok.AllArgsConstructor
import lombok.Data
import lombok.NoArgsConstructor

@Data
@NoArgsConstructor
@AllArgsConstructor
class UserInfoResp {

    @SerializedName("user_id")
    var uin= 0L

    @SerializedName("user_name")
    var name = ""

    @SerializedName("user_displayname")
    var displayName = ""

    @SerializedName("user_remark")
    var remark = ""

    @SerializedName("mail")
    var mail = ""

    @SerializedName("find_method")
    var findMethod = ""

    @SerializedName("max_vote_cnt")
    var maxVoteCnt: Short = 0

    @SerializedName("have_vote_cnt")
    var haveVoteCnt: Short = 0

    @SerializedName("vip_list")
    var vipList: List<VipInfo> = mutableListOf()

    @SerializedName("hobby_entry")
    var hobbyEntry = ""

    @SerializedName("level")
    var level= 0

    @SerializedName("birthday")
    var birthday= 0L

    @SerializedName("login_day")
    var loginDay= 0L

    @SerializedName("vote_cnt")
    var voteCnt= 0L

    @SerializedName("qid")
    var qid = ""

    @SerializedName("is_school_verified")
    var schoolVerified = false

    @SerializedName("location")
    var location: Location? = null

    @SerializedName("cookie")
    var cookie: ByteArray = byteArrayOf()

    @Data
    class VipInfo {
        /**
         * QQ_VIP,
         * SUPER_QQ,
         * SUPER_VIP,
         * QQ_VIDEO,
         * QQ_READING,
         * BIG_VIP,
         * YELLOW_VIP
         */
        var type = ""
        var level= 0

        @SerializedName("vip_type")
        var vipType= 0

        @SerializedName("template_id")
        var templateId= 0L
    }

    @Data
    class Location {
        var city = ""
        var company = ""
        var country = ""
        var province = ""
        var hometown = ""
        var school = ""
    }
}
